n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -379,8 +379,10 @@ def set_replica_weights(self, weights, i_replica=0):
                 the replica number to set, defaulting to 0
         """
         self.get_layer(f"{NN_PREFIX}_{i_replica}").set_weights(weights[NN_PREFIX])
-        self.get_layer(f"{PREPROCESSING_PREFIX}_{i_replica}").set_weights(
-            weights[PREPROCESSING_PREFIX]
+        set_layer_replica_weights(
+            layer=self.get_layer(PREPROCESSING_PREFIX),
+            weights=weights[PREPROCESSING_PREFIX],
+            i_replica=i_replica,
```

Comment:
Instead of doing it in this way, which breaks down the layer into smaller layers, it might be more convenient to do the following:

1. Get all weights, now you have a numpy array
2. Modify the subset that you actually need to modify.
3. Set all weights.

You are getting and resetting the same bunch of numbers for all replicas, it is true, but setting weights by replica will only happen once per fit so the performance hit is more than acceptable. This greatly simply the code and you avoid playing with fire...

Reply:
Thanks, good idea, it seems to work!